docs/DEVGUIDE.md

```diff
@@ -83,13 +84,26 @@ To deploy to Kubernetes, see the
 
 ## Testing
 
-Currently, we only have unit testcases within this repo:
+There are two types of tests: unit and end-to-end (e2e). The unit testcases
```

Comment:
This is what I would call an integration test rather than an e2e.  My definitions of these terms:

1.  Unit - code that tests a particular compilation unit, ie something you can run with `go test`
2.  Integration - starts a component(s) in isolation in order to test that component at a higher level than unit or test an integration between two components, but without verifying the behavior of the entire system
3.  E2E - start a 'full' system and verify the work of the system in a holistic way